force-app/main/default/classes/tasks/CollectionsTasks.cls

```diff
@@ -26,6 +41,12 @@ public class CollectionsTasks {
     Integer sum,
     List<Integer> listWithElementsToCheck
   ) {
+
+    for (Integer i = 0; i < listWithElementsToCheck.size()-1; i++) {
+      if (listWithElementsToCheck[i]+listWithElementsToCheck[i+1] == sum) {
```

Comment:
The task is similar to the previous task.
In the previous task you've tried all the combinations.

This implementation looks like a hack for the test data - I'll fix the test data to have random order and the logic won't work.

Reply:
Updated